misc-tools/dj_make_chroot.in

```diff
@@ -396,6 +400,29 @@ if [ "$DISTRO" = 'Ubuntu' ]; then
     in_chroot "ln -s /bin/true /sbin/initctl"
 fi
 
+# Add additional repository lists to the chroot for upstream repos
+if [ -n "$REPOLISTS" ]; then
+    SPLIT_LIST=$(printf %s "$REPOLISTS" | tr ',' ' ')
+    for provided_repo in $SPLIT_LIST
+    do
+        cat "$provided_repo" >> "$CHROOTDIR/etc/apt/sources.list.d/upstream_chroot.list"
+        for filetype in arm gpg; do
+            if [ -f "${provided_repo}.${filetype}" ]; then
+                GPGFILENAME=$(mktemp -p "$CHROOTDIR/etc/apt/trusted.gpg.d" 'upstream_gpg_XXXXXX.gpg')
+                file_head=$(head -n1 "${provided_repo}.${filetype}")
+                if [ "-----BEGIN PGP ARMORED FILE-----" = "$file_head" ]; then
+                    gpg --dearmour "${provided_repo}.${filetype}"
+                    provided_repo=${provided_repo}.${filetype}
+                fi
+                cp "${provided_repo}.gpg" "$GPGFILENAME"
```

Comment:
I think that it's clearer not to modify the `provided_repo` variable and the suffix a hardcoded `.gpg`, but just define a new variable `repo_sign_key` above and then modify that in the if statement.

Reply:
Thats now implemented